components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt

```diff
@@ -301,12 +300,17 @@ internal class SaveLoginDialogFragment : PromptDialogFragment() {
             username = username,
             password = password
         )
-        // The below code is not thread safe, so we ensure that we block here until we are sure that
-        // the previous call has been canceled
-        runBlocking { validateStateUpdate?.cancelAndJoin() }
-        var validateDeferred: Deferred<Result>? = null
-        validateStateUpdate = scope.launch(IO) {
-            val validationDelegate = feature?.loginValidationDelegate ?: return@launch
+
+        try {
+            validateStateUpdate?.cancelAndJoin()
+        } catch (cancellationException: CancellationException) {
+            Logger.error("Failed to cancel save login dialog", cancellationException)
```

Comment:
nit - we're not cancelling the dialog, we're cancelling the validateStateUpdate job

Reply:
I'm don't even think we need/want to catch this. Looking at the docs for `CancellationException` - 
 * Thrown by cancellable suspending functions if the [Job] of the coroutine is cancelled while it is suspending.
 * It indicates _normal_ cancellation of a coroutine.
 * **It is not printed to console/log by default uncaught exception handler**.